### PR TITLE
Add country support

### PR DESCRIPTION
**Pull request by rsimari**
https://app.asana.com/0/1185123266141337/1191525787597415/f

### CODE REVIEW

File: README.md
Comment (gurpreetatwal) [inline README.md:123]: for the latter (country code description) should we just put that in the doc center? we need to specify somewhere that the only country supported atm is US

Comment (rsimari) [reply README.md:123]: definitely in api-reference, we should probably include that here too

File: README.md
Comment (gurpreetatwal) [inline README.md:37]:
```suggestion
		Country:      "<Country Code>",
```

I _think_ the current code is using tabs, not spaces

File: README.md
Comment (gurpreetatwal) [inline README.md:128]:
```suggestion
	Country: "<Country Code>",
```

File: authentication_test.go
Comment (gurpreetatwal) [inline authentication_test.go:90]: should we a test where we don't specify country?

File: smartcar.go
Comment (gurpreetatwal) [inline smartcar.go:106]: we should default it to `US` if not passed to match the behavior in the other SDKs

File: README.md
Comment (s-ashwinkumar) [inline README.md:37]: @allisonc07 @rsimari 
Sorry for the delayed comment but I also looked over ruby SDK i have a question. There is a difference in what we pass during the auth. Here we expect developers to pass something like `flags: 'country:US'` where as in go SDK we just added an additional parameter `country`. I am assuming you guys did the other SDKs as well, so is it like different for different SDKs ? I think t is worth it to keep th signature somewhat similar, WDYT ?

**Comment by s-ashwinkumar** [comment]
a bunch of places have alignment issues, i am not sure whats going on there and wht the linter didnt catch it. Please look over it, these are places with `country` being used

File: authentication_test.go
Comment (s-ashwinkumar) [reply authentication_test.go:90]: just seeing all your comments. I think i repeated them, that said, i held back on this one. That is because i dont think we have written tests for optional parameters before this either (I for sure havent in ruby). Not sure if we should change that.] and atleast unit tests.

File: README.md
Comment (gurpreetatwal) [reply README.md:37]: I was unsure on this as well, my original assumption was that we'd do something like `flags: 'country:US'` so that way we wouldn't need to do a breaking change to the SDKs when decide to promote `country` from being a flag to a top level parameter.

File: authentication_test.go
Comment (gurpreetatwal) [reply authentication_test.go:90]: I'm fine either way, I mostly just added this because I don't know go and wanted to be sure that optional params in structs work hahah

File: README.md
Comment (allisonc07) [reply README.md]: For the other SDKs (api + connect) that I worked on, I have been expecting the `flags` parameter as a space separated list. I thought that that might be the best approach since it would require the least amount of maintenance if we were to add new flags. 

In terms of defaulting, I think we have decided to default to 'US' in the other SDKs, so we should change that here as well!

**Review by gurpreetatwal — APPROVED** [review]
LGTM, I'd wait for an approval from @allisonc07 or @s-ashwinkumar before merging

File: authentication.go
Comment (allisonc07) [inline authentication.go:49]: in the other SDKs I made `Flags` a `[]string`, so we might want to change to this as well here!

**Review by allisonc07 — COMMENTED** [review]
left a small comment about the type of `Flags`, but the rest looks good to me!

**Review by gurpreetatwal — APPROVED** [review]

**Review by allisonc07 — APPROVED** [review]

**Pull request merged**
